a5py/routines/plotting.py

```diff
@@ -587,13 +592,16 @@ def mesh2d(x, y, z, log=False, diverging=False, xlabel=None, ylabel=None,
         clim[1] = np.nanmax(z)
     z = np.ma.masked_invalid(z)
 
-    if log:
+    if logscale:
         if diverging:
             if cmap == None: cmap = "bwr"
             norm = mpl.colors.SymLogNorm(linthresh=10, linscale=1.0,
                                  vmin=clim[0], vmax=clim[1], base=10)
         else:
             if cmap == None: cmap = "viridis"
+            temp = z[~np.isnan(z)]
```

Comment:
Weird, I just got notifications for these... I could swear I tried without any masking but maybe I'm just misremembering...
I can only have the time for testing this on friday, and I'll try looking into it then.

Reply:
Alright. So I checked it again. The reason I put this there is I not only needed to mask negative and Nan values, but also the zeros. (The lognorm didn't like zeros since e^(-infty) rightarrow 0).

Reply:
My bad! I just now realised that the `temp` is not plotted but it is just a helper quantity to calculate `clim`. Replace that with a one-liner `if clim[0] <=0: clim[0] = np.min(z[z>0])`. Note the `ìf` here as without it you'd always overwrite the user-specified value.

Just commit that change and then you are free to merge.